test/setup_helpers.go

```
@@ -0,0 +1,17 @@
+package test
+
+import "net"
+
+// GetFreePort determines a free port on the host. It does so by delegating the job to net.ListenTCP.
+// Then providing a port of 0 to net.ListenTCP, it will automatically choose a port for us.
+func GetFreePort() (int, error) {
+	if a, err := net.ResolveTCPAddr("tcp", "localhost:0"); err == nil {
+		var l *net.TCPListener
+		if l, err = net.ListenTCP("tcp", a); err == nil {
+			port := l.Addr().(*net.TCPAddr).Port
+			err = l.Close()
+			return port, err
+		}
+	}
+	return -1, nil
```

Please return an error here.

---

Done